# КОММЕНТАРИЙ РЕВЬЮВЕРА
Привет Денис👋!)

Меня зовут Сибирев Владислав, я осуществлял проверку проекта. Может перейдем сразу на <b>ты</b>? Если это неприемлемо, то обязательно дай знать)

Далее по ходу работы ты сможешь встретить мои комментарии, они будут отмечены разными цветами:

<div class="alert alert-block alert-success">
 ✔️ Похвала и возможности развития
</div>
<div class="alert alert-block alert-warning">
⚠️ Некритичные замечания, но стоит обратить внимание
</div>
<div class="alert alert-block alert-danger">
❌ Грубая ошибка, необходимо исправить
</div>


Заранее прошу не удалять мои комментарии, это важно. Если у тебя возникнут вопросы, обозначь их, я с радостью тебе помогу!)

Как вариант, можешь обозначить их так:
<div class="alert alert-block alert-info">
<b>КОММЕНТАРИЙ СТУДЕНТА:</b> Здравствуй. Первый тестовый вариант отправил без форматирования, поторопился, не знал как это работает. Сейчас привожу в удобный вид. 
    Print() печатаю(копирую) из Питона-черновика, в финальной версии стилистика будет соблюдена.
</div>

# Исследование надежности заемщиков


Во второй части проекта вы выполните шаги 3 и 4. Их вручную проверит ревьюер.
Чтобы вам не пришлось писать код заново для шагов 1 и 2, мы добавили авторские решения в ячейки с кодом. 



<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b> 

✔️ Так как первая часть является автоматической, то я просто буду в ней давать рекомендации, которые ты можешь использовать в дальнейшем))<br>
P.S. Любой проект нужно начинать с описания)) 
  </div>

## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [150]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b> 

✔️ Молодец, что используешь конструкцию __try и except__<br>
P.S. Для следующих проектов нужно импортировать библиотеки в отдельной ячейке) Это позволяет быстро сориентироваться, что нужно для запуска проекта, если кто-то будет запускать его в другом окружении.
  </div>

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [151]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [152]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [153]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

✔️ Изучить долю пропущенных пропусков можно еще с помощью `data.isna().mean()*100`
  </div>

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [154]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

 ✔️  Можно также использовать следующую конструкцию `data['total_income'] = data['total_income'].fillna(data.groupby(['income_type'])['total_income'].transform('median'))`
Функция transform() в принципе полезная https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.transform.html
  </div>

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [155]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [156]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [157]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [158]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

 ✔️  Все верно, можно еще использовать query(), который позволяет работать с нужными данными) https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.query.html
  </div>

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [159]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [160]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [161]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [162]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру.**

In [163]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [164]:
data.duplicated().sum()

71

In [165]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [166]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [167]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

✔️  Функцию написал правильную, молодец, но применение функции лучше проверять с помощью `value_counts`, вдруг какое то условие не сработало)

  </div>

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [168]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [169]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [170]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [171]:
child_dep=(data.groupby('children')['debt']).agg(['count', 'sum', 'mean']) #создал промежуточный датафрейм для вычислений
child_dep['mean'] = child_dep['mean'].map('{:.1%}'.format) #привёл проценты в удобный вид
child_dep.rename(columns={'count': 'Всего заёмщиков', 'sum': 'Просрочек по выплатам', 'mean': 'Соотношение'}, inplace=True) #переименовал столбцы для удобства
child_dep
# Ваш код будет здесь. Вы можете создавать новые ячейки.

Всего заёмщиков  Просрочек по выплатам Соотношение
children                                                    
0                   14091                   1063        7.5%
1                    4808                    444        9.2%
2                    2052                    194        9.5%
3                     330                     27        8.2%
4                      41                      4        9.8%
5                       9                      0        0.0%

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюер:</b> 

⚠️ Все отлично по поводу таблиц, но с помощью `Print()`  не нужно выводить таблицs, просто название прописываешь таблицы и все выведется<br> Это касается всех таблиц)

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️  Здесь все отлично стало)) 

  </div>

**Вывод:**

Проведённый анализ показывает, что наименьший процент по просрочкам - **7.5%** - `у заёмщиков без детей`. 

У заёмщиков с `одним` или `двумя` детьми ситуация хуже - **9.2%** и **9.4%** соответственно.
Количество заёмщиков, имеющих троих и более детей, отличаются на порядок от предыдущих групп и сравнивать их некорректно.

Зависимость между количеством детей и возвратом кредита в срок - `существует`.


<div class="alert alert-block alert-warning">
<b>Комментарий ревьюер:</b> 

⚠️ Не бойся использовать в выводах цифры, также нужно сделать акцент на сбалансированности выборок. Примерно выводы будут выглядеть так: 
`люди, которые не имеют детей, лучше всего отдают кредит, доля должников здесь составляет с 7,5%, хуже обстоит дело с кредиторами, кто имеет одного или двух детей, здесь доля невозврата составляет соответственно 9,2% и 9,4%. Остальные категории уже некорректно сравнивать, так как в них очень маленькая выборка`)) <br>

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️  Выводы переработал, молодец)) 

  </div>

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [176]:
data.loc[data['family_status'] == 'вдовец / вдова', 'family_status'] = 'одинок' #объединяю 2 группы в 1
data.loc[data['family_status'] == 'в разводе', 'family_status'] = 'одинок' #объединяю 2 группы в 1

family_dep = (data.groupby('family_status')['debt']).agg(['count','sum', 'mean'])
family_dep['mean'] = family_dep['mean'].map('{:.1%}'.format) #привёл проценты в удобный вид
family_dep.rename(columns={'count': 'Всего заёмщиков', 'sum': 'Просрочек по выплатам', 'mean': 'Соотношение'}, inplace=True) #переименовал столбцы для удобства
family_dep.sort_values(by='Соотношение', ascending=True) #сортировка по возрастанию и вывод на экран

# Ваш код будет здесь. Вы можете создавать новые ячейки.

Всего заёмщиков  Просрочек по выплатам Соотношение
family_status                                                            
одинок                            2140                    147        6.9%
женат / замужем                  12261                    927        7.6%
гражданский брак                  4134                    385        9.3%
Не женат / не замужем             2796                    273        9.8%

**Вывод:**

Просматривается зависимость, говорящая, что люди после брака(`развод, потеря`) с большей вероятностью возращают кредит в срок - **6.9%**. 

Затем люди, состоящие в браке: `официальный` - **7.6%** и `гражданский` - **9.3%**. 

Хуже всего с возвратом кредита в срок дела обстоят у `не женатых\не замужних` - **9.8%**.

Зависимость между семейным положением и возвратом кредита в срок - `существует`.

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюер:</b> 

⚠️ В выводах давай будет использовать цифры, также  удели здесь внимание тоже сбалансированности выборок))

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️  Здесь тоже все отлично по выводам))

  </div>

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [89]:
income_dep = (data.groupby('total_income_category')['debt']).agg(['count','sum', 'mean'])
income_dep['mean'] = income_dep['mean'].map('{:.1%}'.format) #привёл проценты в удобный вид
income_dep.rename(columns={'count': 'Всего заёмщиков', 'sum': 'Просрочек по выплатам', 'mean': 'Соотношение'}, inplace=True) #переименовал столбцы для удобства
income_dep.sort_values(by='Всего заёмщиков', ascending=False) #сортировка и вывод на экран

# Ваш код будет здесь. Вы можете создавать новые ячейки.

Всего заёмщиков  Просрочек по выплатам Соотношение
total_income_category                                                    
C                                15921                   1353        8.5%
B                                 5014                    354        7.1%
D                                  349                     21        6.0%
A                                   25                      2        8.0%
E                                   22                      2        9.1%

**Вывод:**

Учитывая, что в группы B и C входят более 98% человек, рассмотрим только их.

У заёмщиков с доходом от 50001 до 200000 (`Группа C`) процент просрочек выше - **8.5%**.

У заёмщиков с доходом от 200001 до 1000000 (`Группа B`) процент просрочек лучше - **7.1%**.

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюер:</b> 

⚠️ Здесь сравниваем только две группы B и С, не забываем про сбалансированость выборки))<br>

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️  Здесь сравнил две самые большие группы))

  </div>

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [92]:
purpose_dep = (data.groupby('purpose_category')['debt']).agg(['count','sum', 'mean'])
purpose_dep['mean'] = purpose_dep['mean'].map('{:.1%}'.format) #привёл проценты в удобный вид
purpose_dep.rename(columns={'count': 'Всего заёмщиков', 'sum': 'Просрочек по выплатам', 'mean': 'Соотношение'}, inplace=True) #переименовал столбцы для удобства
purpose_dep.sort_values(by='Соотношение', ascending=True) #сортировка и вывод на экран

# Ваш код будет здесь. Вы можете создавать новые ячейки.

Всего заёмщиков  Просрочек по выплатам Соотношение
purpose_category                                                            
операции с недвижимостью            10751                    780        7.3%
проведение свадьбы                   2313                    183        7.9%
операции с автомобилем               4279                    400        9.3%
получение образования                3988                    369        9.3%

**Вывод:**

Меньше всего просрочек по `операциям с недвижимостью` - **7.3%**, что скорее всего связанно с особой ценностью собственого жилья, за которое выплачивается ипотека. В эту категорию так же включены действия с коммерческой недвижимостью. Вероятно, процент говорит о высокой ответственности за погашение взятого кредита. 

Процент просрочек кредитов `на проведение свадьбы` выше - **7.9%**.

Самый большой процент - с кредитами `на получение образование` и `операции с автомобилем` - **9.3%**.

<div class="alert alert-block alert-warning">
<b>Комментарий ревьюера:</b> 

⚠️ Почему по твоему самые надежные заемщики те, кто берет ипотеку? Здесь можно порасуждать в выводах)) 

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️ Логика твоя верная, самые большие суммы при ипотеке, поэтому банки предъявляют самые жесткие требования к ипотечникам, поэтому их проверяют вдоль и поперек при выдаче кредитов, плюс собственность находится в залоге у банка, не будешь платить она вернется к банку)) 
   
       
  </div>

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* Пропуски в данных могут появляться из-за человеческого фактора или технической неисправности. 

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

✔️ Ответ правильный)) 
  </div>

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Медианное значение не восприимчиво к аномальным значениям (слишком большим или слишком маленьким от среднего).

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>

✔️ Медиана не подвержена выбросам, все верно)) 

### Шаг 4: общий вывод.

Проведенный анализ 21331 выданного кредита показывает качества заёмщика, который с высокой вероятностью вернёт кредит в срок:<br>
Заёмщик `без детей` вернет кредит в срок с вероятностью **92.5%**<br>
Заёмщик, `ранее состоявший в браке` возращает кредит в срок с вероятностью **93.1%**<br>
В случае если кредит берётся для `операций с недвижимостью` - **92.7%**<br>
При зарплата от `50000 до 200000` - процент возврата кредита в срок равен **91.5%**<br>
<br>
<br>
Для лучшего анализа данных в будущем рекомендуется приводить данные к единому образцу, исключать пропуски и опечатки.

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>

✔️ Выводы я бы все таки сделал более детальными, они все таки должны состоять из промежуточных выводов)) Я дам тебе ссылки на полезные гайды, которые могут помочь при оформлении текстов:
    
<a href="https://markdown-guide.readthedocs.io/en/latest/basics.html">guide1.</a>
<a href="https://paulradzkov.com/2014/markdown_cheatsheet/">guide2.</a>
    
А также советую тебе взглянуть на <a href="https://www.kaggle.com/notebooks/">работы</a> профессиональных аналитиков, чтобы оценить, как другие люди оформляют и проводят исследования.
   
       
  </div>

<div class="alert alert-block alert-danger">
<b>Комментарий ревьюера:</b> 

❌ Выводы должны быть детальными:
- Отвечаешь на вопрос "Есть ли зависимость между количеством детей и возвратом кредита в срок?"
- Отвечаешь на вопрос  "Есть ли зависимость между семейным положением и возвратом кредита в срок?"
- Отвечаешь на вопрос  "Есть ли зависимость между уровнем дохода и возвратом кредита в срок?"
- Отвечаешь на вопрос  "Как разные цели кредита влияют на его возврат в срок?"
- Даешь рекомендацией банку по улучшению сбора данных и скоринга.

Общий вывод - резюме этой работы. Большинство заказчиков посмотрят сначала его (а некоторые только его). Чтобы выводы не огорошивали заказчика своей очевидностью, стоит добавить в них цифр, всяких корреляций, процентов итд :) Сделай так, чтобы вся найденная в данных информация была отражена в выводе))

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера:</b>
    
Денис, отлично поработал, видно что умеешь работать с кодом, в общем молодец, но давай красиво выведем таблицы и поработаем над выводами)) 
<br> Жду исправленного варианта)
  </div>

<div class="alert alert-block alert-success">
<b>Комментарий ревьюера_V2:</b>
    
Дения, отлично поработал, можно двигаться дальше)) Успехов в дальнейшем обучении)) 
  </div>